File: app/src/main/java/com/saienko/androidthings/barman/db/AppDatabase.java
package com.saienko.androidthings.barman.db;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;
import com.saienko.androidthings.barman.db.cocktail.*;
import com.saienko.androidthings.barman.db.drinkGroup.CocktailGroup;
import com.saienko.androidthings.barman.db.drinkGroup.CocktailGroupDao;
import com.saienko.androidthings.barman.db.gpio.Gpio;
import com.saienko.androidthings.barman.db.gpio.GpioDao;
import com.saienko.androidthings.barman.db.motor.Motor;
import com.saienko.androidthings.barman.db.motor.MotorDao;
import com.saienko.androidthings.barman.db.position.Position;
import com.saienko.androidthings.barman.db.position.PositionDao;

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/9/17
 * Time: 13:34
 */

@Database(entities = {Motor.class, Gpio.class, Cocktail.class, CocktailGroup.class, Component.class,
                      com.saienko.androidthings.barman.db.cocktail.CocktailElement.class,
                      Position.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    private static AppDatabase instance;

    public abstract MotorDao motorDao();

    public abstract GpioDao gpioDao();

    public abstract CocktailDao cocktailDao();

    public abstract CocktailGroupDao cocktailGroup();

    public abstract ComponentDao componentDao();

    public abstract CocktailElementDao cocktailElementDao();

    public abstract PositionDao positionDao();

    public static void init(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "db_v1.db").build();
        } else {
            throw new UnsupportedOperationException("Unknown action");
        }
    }

    public static AppDatabase getDb() {
        return instance;
    }


}

File: app/src/main/java/com/saienko/androidthings/barman/db/AppDatabase.kt
package com.saienko.androidthings.barman.db

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import com.saienko.androidthings.barman.db.dao.*
import com.saienko.androidthings.barman.db.entity.*

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/9/17
 * Time: 13:34
 */

@Database(
        entities = [(Motor::class), (Gpio::class), (Cocktail::class), (Component::class), (CocktailElement::class), (Position::class)],
        version = 1,
        exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun motorDao(): MotorDao

    abstract fun gpioDao(): GpioDao

    abstract fun cocktailDao(): CocktailDao

    abstract fun componentDao(): ComponentDao

    abstract fun cocktailElementDao(): CocktailElementDao

    abstract fun positionDao(): PositionDao

    companion object {
        const val DATABASE_NAME = "alco"
    }

//    companion object {
//        private lateinit var context: Context
//        private val database: AppDatabase by lazy(LazyThreadSafetyMode.SYNCHRONIZED) {
//            Room.databaseBuilder(context, AppDatabase::class.java, "alco.db").build()
//        }
//
//        fun init(context: Context) {
//            this.context = context.applicationContext
//        }
//
//        fun getDb(): AppDatabase {
//            return database
//        }
//
//    }
}